docs/docs/development/backend/exception.md

```diff
@@ -8,38 +8,40 @@
 Various errors can occur in the backend, which must be made understandable to
 the end user and the developers when calling the API.
 
-In general, please use the following syntax to return a error message from the
-backend:
+In order to maintain consistency and benefit from automatic exception
+registration, define exceptions in a route-specific exceptions.py file in the
+following manner:
 
-```py title="routes.py"
-from fastapi import HTTPException
+```py title="exceptions.py"
+from fastapi import status
+from capellacollab.core import exceptions as core_exceptions
 
-raise HTTPException(
-    status_code=403,  # (1)
-    detail={
-        "err_code": "project_deletion_permission_denied",  # (2)
-        "title": "Permission denied",  # (3)
-        "reason": (
-            "You don't have the permission to delete projects.",
-            "Please ask your administrator to delete the project.",
-        ),  # (4)
-        "technical": "The role administrator is required.",  # (5)
-    },
-)
+class UserNotFoundError(core_exceptions.BaseError): # (1)
```

Comment:
The `# (1)` looks like you wanted to use code annotations, but doesn't render / resolve properly:
![image](https://github.com/DSD-DBS/capella-collab-manager/assets/23395732/7c6b14da-f486-49a9-b7f6-09ce6f55b20c)

More info: 
https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-annotations & https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#code-annotations

Reply:
This was just a straight update of the existing docs, which uses the numbers as indicators to a section below. Do you want the annotations hidden?
<img width="1846" alt="main_vs_branch_docs" src="https://github.com/DSD-DBS/capella-collab-manager/assets/79602065/ce73f67a-6503-46d9-aa76-d2154e0da0f6">

Reply:
I had to update the error code part to _required_ anyway, so I went ahead and made them annotations.